### PR TITLE
chore(react): add deprecation notice to react page

### PR DESCRIPTION
**Pull request by annawen1**
### Related Ticket(s)

[react]: add deprecation notice to package #9134

### Description

Add notice for react package deprecation

changes viewable here: [https://ibmdotcom-library.s3-web.us-south.cloud-object-storage.appdomain.cloud/deploy-previews/1502/developing/frameworks/react/index.html](https://ibmdotcom-library.s3-web.us-south.cloud-object-storage.appdomain.cloud/deploy-previews/1502/developing/frameworks/react/index.html)

### Changelog

**New**

- {{new thing}}

**Changed**

- {{changed thing}}

**Removed**

- {{removed thing}}

### CODE REVIEW

File: src/pages/developing/frameworks/react.mdx [hunk 1]
```diff
@@ -6,6 +6,12 @@ tabs: ['Web Components', 'React']
 
 import { Link } from 'gatsby';
 
+<InlineNotification>
+
+**Note:** The Carbon for IBM.com React package will be deprecated in Carbon for IBM.com V2. This package will fall into maintaince-mode only; we will make updates only for potential security issues and high severity bugs. For an estimated timeline on the V2 release, you can view the Carbon for IBM.com [roadmap](https://github.com/carbon-design-system/carbon-for-ibm-dotcom#workspaces/carbon-for-ibmcom-5d449f3642eb1962336cbe52/roadmap). With Carbon for IBM.com V2, the Carbon for IBM.com Web Components package will be the main package moving forward. Applications using React can use the [react-wrapper](/developing/frameworks/web-components#react-wrapper) components supported by the the Carbon for IBM.com Web Components package.
```
Comment (kennylam) [inline src/pages/developing/frameworks/react.mdx:11]:
```suggestion
**Note:** The Carbon for IBM.com React package will be deprecated in Carbon for IBM.com V2. This package will fall into maintaince-mode, with updates only for potential security issues and high severity bugs. For an estimated timeline on the V2 release, you can view the Carbon for IBM.com [roadmap](https://github.com/carbon-design-system/carbon-for-ibm-dotcom#workspaces/carbon-for-ibmcom-5d449f3642eb1962336cbe52/roadmap). In Carbon for IBM.com V2, [@carbon/ibmdotcom-web-components](https://www.npmjs.com/package/@carbon/ibmdotcom-web-components) will be the main package moving forward. Applications using React can use the [react-wrapper](/developing/frameworks/web-components#react-wrapper) components supported by the the Carbon for IBM.com Web Components package.
```

File: src/pages/developing/frameworks/react.mdx [hunk 2]
```diff
@@ -6,6 +6,12 @@ tabs: ['Web Components', 'React']
 
 import { Link } from 'gatsby';
 
+<InlineNotification>
```
Comment (kennylam) [inline src/pages/developing/frameworks/react.mdx:9]:
```suggestion
<InlineNotification kind="warning">
```

@annawen1 Do we want to try a different notification type?

@oliviaflory @RichKummer unless you guys wanna stick with this default `info` one.

Comment (annawen1) [reply src/pages/developing/frameworks/react.mdx:9]: hmmm I think warning is good!

**Review by kennylam — APPROVED** [review]
LGTM!

**Review by oliviaflory — APPROVED** [review]
LGTM!

**Review by IgnacioBecerra — APPROVED** [review]

**Pull request merged**